auth/src/main/java/com/firebase/ui/auth/ui/idp/WelcomeBackIdpPrompt.java

```diff
@@ -111,93 +111,44 @@ protected void onCreate(Bundle savedInstanceState) {
             return;
         }
 
-        ((TextView) findViewById(R.id.welcome_back_idp_prompt))
-                .setText(getIdpPromptString(oldUser.getEmail()));
+        ((TextView) findViewById(R.id.welcome_back_idp_prompt)).setText(getString(
+                R.string.fui_welcome_back_idp_prompt,
+                existingUser.getEmail(),
+                getString(mProvider.getNameRes())));
 
-        mIdpProvider.setAuthenticationCallback(this);
         findViewById(R.id.welcome_back_idp_button).setOnClickListener(new OnClickListener() {
             @Override
             public void onClick(View view) {
-                getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_signing_in);
-                mIdpProvider.startLogin(WelcomeBackIdpPrompt.this);
+                mProvider.startLogin(WelcomeBackIdpPrompt.this);
             }
         });
-    }
-
-    private String getIdpPromptString(String email) {
-        return getString(R.string.fui_welcome_back_idp_prompt, email, mIdpProvider.getName(this));
-    }
 
-    @Override
-    public void onActivityResult(int requestCode, int resultCode, Intent data) {
-        super.onActivityResult(requestCode, resultCode, data);
-        mIdpProvider.onActivityResult(requestCode, resultCode, data);
-    }
+        handler.getOperation().observe(this, new Observer<Resource<IdpResponse>>() {
+            @Override
+            public void onChanged(Resource<IdpResponse> resource) {
+                if (resource.getState() == State.LOADING) {
+                    getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_loading);
+                    return;
+                }
+                getDialogHolder().dismissDialog();
 
-    @Override
-    public void onSuccess(@NonNull final IdpResponse idpResponse) {
-        AuthCredential newCredential = ProviderUtils.getAuthCredential(idpResponse);
-        if (newCredential == null) {
-            throw new IllegalStateException("Unknown provider: " + idpResponse.getProviderType());
-        }
+                if (resource.isUsed()) { return; }
```

Comment:
Maybe if we're going to have this logic everywhere we could instead make our own `Observer` subclass to handle it?

It could have separate callbacks for loading, and first-success or first-finish. 

Whatever we decide, I just want to get somewhere that we're not randomly adding this used check because I know one day we'll forget it and get some rotation bugs again